mkosi/machine.py

```diff
@@ -31,16 +36,30 @@
 
 
 class Machine:
-    def __init__(self, args: Optional[Sequence[str]] = None, debug: bool = False) -> None:
-        # Remains None until image is built and booted, then receives pexpect process
+    def __init__(self, args: Sequence[str] = [], debug: bool = False) -> None:
+        # Remains None until image is built and booted, then receives pexpect process.
         self._serial: Optional[pexpect.spawn] = None
         self.exit_code: int = -1
         self.debug = debug
         self.stack = contextlib.ExitStack()
         self.args: MkosiArgs
 
-        # We make sure to add the arguments in the machine class itself, rather than typing this for every testing function.
         tmp = parse_args(args)["default"]
+
+        # By default, Mkosi makes Verb = build.
+        # But we want to know if a test class passed args without a Verb.
+        # If so, we'd like to assign default values or the environment's variable value.
+        if tmp.verb == Verb.build and Verb.build.name not in args:
+            default_verb = os.getenv("MKOSI_TEST_DEFAULT_VERB")
```

Comment:
Nit: for local variables, there's no need to bother with long variable names, the code is perfectly readable if this variable is just named `verb`, no need to go for a long name such as `default_verb`